engine/resource/src/resource.cpp

```diff
@@ -1070,119 +1154,90 @@ static Result DoGet(HFactory factory, const char* name, void** resource)
 
     char extbuffer[64];
     const char* ext = GetExtFromPath(canonical_path, extbuffer, sizeof(extbuffer));
-    if (ext)
+
+    if (!ext)
     {
-        ext++;
+        dmLogWarning("Unable to load resource: '%s'. Missing file extension.", canonical_path);
+        return RESULT_MISSING_FILE_EXTENSION;
+    }
 
-        SResourceType* resource_type = FindResourceType(factory, ext);
-        if (resource_type == 0)
-        {
-            dmLogError("Unknown resource type: %s", ext);
-            return RESULT_UNKNOWN_RESOURCE_TYPE;
-        }
+    // skip over '.'
+    ext++;
 
-        void *buffer;
-        uint32_t file_size;
-        Result result = LoadResource(factory, canonical_path, name, &buffer, &file_size);
-        if (result != RESULT_OK) {
-            if (result == RESULT_RESOURCE_NOT_FOUND) {
-                dmLogWarning("Resource not found: %s", name);
-            }
-            return result;
-        }
+    SResourceType* resource_type = FindResourceType(factory, ext);
+    if (resource_type == 0)
+    {
+        dmLogError("Unknown resource type: %s", ext);
+        return RESULT_UNKNOWN_RESOURCE_TYPE;
+    }
 
-        assert(buffer == factory->m_Buffer.Begin());
+    *resource_type_out = resource_type;
 
-        // TODO: We should *NOT* allocate SResource dynamically...
-        SResourceDescriptor tmp_resource;
-        memset(&tmp_resource, 0, sizeof(tmp_resource));
-        tmp_resource.m_NameHash = canonical_path_hash;
-        tmp_resource.m_ReferenceCount = 1;
-        tmp_resource.m_ResourceType = (void*) resource_type;
+    return RESULT_OK;
+}
 
-        void *preload_data = 0;
-        Result create_error = RESULT_OK;
+// Assumes m_LoadMutex is already held
+static Result CreateResourceFromDisk(HFactory factory, const char* name, void** resource)
```

Comment:
Is this strictly from "disk" or is it just another helper function?
The DoLoadResourceLocked also support http, so I wouldn't call it "from disk".

Reply:
Ah right, yes I renamed it.